cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -408,7 +410,8 @@ private RejectedExecutionHandler createAsyncRejectedExecutionHandler() {
       @Override
       public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
         if (!executor.isShutdown()) {
-          metricsCollector.increment("collect.async.reject", 1);
+          metricsCollector.childCollector(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE)
```

Comment:
Thanks for that tip Terence. 

We would however need to cache it for the ``createStreamMetricsCollectionFactory()`` method. 
Actually, I think while namespacing streams, we will have to refactor the metrics collectors in this class, since ``namespaceId`` will not be constant. I've created a note for myself, and will fix it as part of [CDAP-1244](https://issues.cask.co/browse/CDAP-771)